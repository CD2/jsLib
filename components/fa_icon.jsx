import React from 'react'
import PropTypes from 'prop-types'
import {styled} from "utils/theme/index"
import decorate from 'utils/decorate'

export default class FaIcon extends React.Component {

  static PropTypes = {
    icon: PropTypes.string.isRequired,
    hoverColor: PropTypes.string,
    color: PropTypes.string,
    size: PropTypes.number,
    onClick: PropTypes.func,
  }

  get className() {
    let { className, icon } = this.props
    className += ` fa fa-${icon}`
    return className
  }

  render() {
    const { onClick } = this.props
    return (
      <i className={this.className} onClick={onClick}/>
    )
  }
}
export default decorate(
  styled`
    ${({ color, theme, size }) => {
      const col = color || theme.darkBackground
      const sze = size || 1
      return (
        `
        color: ${col};
        font-size: ${sze}em;
        `
      )
    }}
    ${({ hoverColor }) => {
      if(hoverColor) {
        return (
          `
          &:hover{
            color: ${hoverColor};
          }
          `
        )
      }
    }}
  `,
  FaIcon
)
