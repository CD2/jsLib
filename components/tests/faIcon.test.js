/* eslint-disable react/jsx-filename-extension */
import React from 'react'
import { faIcon } from '../fa_icon'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe(`<faIcon />`, () => {
  const view = shallow((
    <faIcon />
  ))

  it(`snapshot`, ()=>expect(toJson(view)).toMatchSnapshot())
})