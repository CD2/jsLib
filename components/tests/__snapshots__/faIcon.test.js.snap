// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<faIcon /> snapshot 1`] = `<faIcon />`;

exports[`snapshot 1`] = `<faIcon />`;
